src/org/rascalmpl/parser/ParserGenerator.java

```diff
@@ -251,6 +252,42 @@ public Class<IGTD<IConstructor, ITree, ISourceLocation>> getNewParser(IRascalMon
 		}
 	}
 
+	/**
+   * Generate a parser from a Rascal grammar and write it to disk
+   * 
+   * @param monitor a progress monitor; this method will contribute 100 work units
+   * @param loc     a location for error reporting
+   * @param name    the name of the parser for use in code generation and for later reference
+   * @param grammar a grammar
+   * @return A parser class, ready for instantiation
+	 * @throws IOException
+   */
+  public void writeNewParser(IRascalMonitor monitor, ISourceLocation loc, String name, IMap definition, ISourceLocation target) throws IOException {
+	String JOB = "Generating parser:" + name;
+	monitor.jobStart(JOB, 100, 60);
+
+	try (OutputStream out = URIResolverRegistry.getInstance().getOutputStream(target, false)) {
+		String normName = name.replaceAll("::", "_").replaceAll("\\\\", "_");
+		monitor.jobStep(JOB, "Generating java source code for parser: " + name,30);
+		IString classString;
+		IConstructor grammar = IRascalValueFactory.getInstance().grammar(definition);
+
+		synchronized (evaluator) {
+			classString = (IString) evaluator.call(monitor, "newGenerate", vf.string(packageName), vf.string(normName), grammar);
+		}
+		debugOutput(classString, System.getProperty("java.io.tmpdir") + "/parser.java");
+		monitor.jobStep(JOB,"Compiling generated java code: " + name, 30);
+		
+		bridge.compileJava(loc, packageName + "." + normName, classString.getValue(), out);
+	} catch (ClassCastException e) {
+		throw new ImplementationError("parser generator:" + e.getMessage(), e);
+	} catch (Throw e) {
+		throw new ImplementationError("parser generator: " + e.getMessage() + e.getTrace());
```

Comment:
You can only test this by having a serious bug in the parser generator. This should never happen.